src/include/txn_inline.h

```diff
@@ -455,19 +485,29 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
              * either set the timestamp on all the updates, or we have set the timestamp on none of
              * the updates.
              */
-            if (*updp != NULL && (*updp)->start_ts == WT_TS_NONE) {
```

Comment:
Should we also only validate if (*updp)->start_ts == WT_TS_NONE?

Reply:
No. This update has generated by this transaction, but the commit timestamp may already updated before the commit. So we have to check every update of this tranaction.

Reply:
ok